prescriptions/_predictable_stacks/ps_cv_torch.yaml

```diff
@@ -0,0 +1,33 @@
+units:
+  wraps:
+  - name: ThothPSCVTorchBaseImageWrap
+    type: wrap
+    should_include:
+      adviser_pipeline: true
+      runtime_environments:
+        base_images: [null]  # Only if no base image is used.
+    match:
+    - state:
+        resolved_dependencies:
+        - name: jupyter-tensorboard
+    - state:
+        resolved_dependencies:
+        - name: opencv-python
+    - state:
+        resolved_dependencies:
+        - name: pillow
+          version: ">8.3.0"
+    - state:
+        resolved_dependencies:
+        - name: tensorboard
+    - state:
+        resolved_dependencies:
+        - name: torch
+    - state:
+        resolved_dependencies:
+        - name: torchvision
+    run:
+      justification:
+      - type: INFO
+        message: Consider using predictive stack for computer vision with Pytorch
```

Comment:
Based on the match part, this message will be shown also when `tensorboard`, `jupyter-tensorboard` or `pillow` will be resolved. Is it expected?

Reply:
Maybe not, maybe only when any of those are combined with the main packages, Pytorch and opencv. I will remove them then. Thanks @fridex !

Reply:
What about adjusting it to make it more generic - not specific to Pytorch, but to a computer vision stack?

Reply:
> Maybe not, maybe only when any of those are combined with the main packages, Pytorch and opencv. I will remove them then. Thanks @fridex !

Actually thinking a little bit more on this, if their software stacks is a subset of the list of packages stated in the prescription, we could recommend a base image (or more in the future maybe, sorted by?), wdyt?

Reply:
The packages can be kept in the prescription as the recommender system can provide suggestions on which base image to use. The justification message can be adjusted to be more generic, not specific to Pytorch. Users can use the container image if it provides used packages.

> if they have have more packages that those ones in the prescription (direct one) we won't recommend it right?, wdyt?

We can still advertise the base images and users can build on top of them. If the intention is to ecommend the base image based on the direct dependencies, you can provide `boot` prescription unit that can act on direct dependencies used in the application - see https://thoth-station.ninja/docs/developers/adviser/prescription/boots.html#boot-match

Reply:
> The packages can be kept in the prescription as the recommender system can provide suggestions on which base image to use. The justification message can be adjusted to be more generic, not specific to Pytorch. Users can use the container image if it provides used packages.
Well this prescription is for Computer Vision with Pytorch image:

```
      justification:
      - type: INFO
        message: Consider using predictive stack for computer vision with Pytorch
        link: https://quay.io/repository/thoth-station/ps-cv-pytorch
```

> 
> > if they have have more packages that those ones in the prescription (direct one) we won't recommend it right?, wdyt?
> 
> We can still advertise the base images and users can build on top of them. If the intention is to ecommend the base image based on the direct dependencies, you can provide `boot` prescription unit that can act on direct dependencies used in the application - see https://thoth-station.ninja/docs/developers/adviser/prescription/boots.html#boot-match

Reply:
> Is this a different prescription to be prepared? Isn't the `type: wrap` doing that purpose?

The same way as Kubernetes provides different objects, such as Job or Deployment, the same way the resolver provides types of pipeline units as an abstraction to configure the resolution process. `wrap` is executed at the end of the resolution process on a fully resolved set of packages whereas `boot` is executed at the beginning of the resolution process on direct dependencies (just for completeness: there are other inputs that they can act on, but that is not relevant for this case).

Now, the difference in `wrap.match` part - this configuration will add the justification mentioned when `jupyter-tensorboard` OR `torch` is present:

```yaml
    match:
    - state:
        resolved_dependencies:
        - name: jupyter-tensorboard
     - state:
        resolved_dependencies:
        - name: torch
```

Whereas this configuration will add the justification when `jupyter-tensorboard` AND `torch` are present in the resolved set of packages at the same time:

```yaml
    match:
    - state:
        resolved_dependencies:
        - name: jupyter-tensorboard
        - name: torch
```

Additional version ranges and index configuration can be applied.

To recommend the base image, it might be good to pick "representatives" that describe the container image the best. This way, we can recommend a container image for computer vision with `pytorch` if `pytorch` is present. On the other hand, `jupyter-tensorboard` might not be a good package to justify recommending pytorch specific computer vision container images.

This might be a good start, then we can perform a more complicated things as discussed in https://github.com/AICoE/recommend-base-image-tutorial/issues/1

Reply:
Thanks for the detailed explanation @fridex, now I understand :) and I think we can start with the `boot` then and if we want to do something more complex later we can make it a warp (maybe when we analyze ps- images and learn specific versions per tags, so we can recommend more precise image tag)

I agree with you on the use of "representatives"! I will modify the PR, thanks!!